src/storage/btree.c

```diff
@@ -11137,7 +11138,8 @@ btree_find_oid_and_its_page (THREAD_ENTRY * thread_p, BTID_INT * btid_int, OID *
   do
     {
       PERF_UTIME_TRACKER_START (thread_p, &ovf_fix_time_track);
-      overflow_page = pgbuf_fix_read_old_and_check_repl_desync (thread_p, overflow_vpid, PGBUF_UNCONDITIONAL_LATCH);
+      overflow_page = pgbuf_fix_old_and_check_repl_desync (thread_p, overflow_vpid, PGBUF_LATCH_WRITE,
```

Comment:
As far as I understand, the purpose of 'pgbuf_fix_read_old_and_check_repl_desync' function was to be run only on PTS for online transactions (which should all be read-only).
Do we want to adapt  'pgbuf_fix_read_old_and_check_repl_desync' to be used for replication (or vacuum after replication) purposes which is the only context where PTS performs writes ?

What if in context of vacuum, the error   ER_PAGE_AHEAD_OF_REPLICATION is ignored, and the page is not vacuumed at all ?

Reply:
I do not think that `pgbuf_fix_read_old_and_check_repl_desync` was meant to be only called on PTS - since in https://github.com/CUBRID/cubrid/pull/3271/files it is in-plare replacement of `pgbuf_fix`.
I think that whether read-write or read-only transactions are allowed on PTS is controlled from a higher layer in the decision stack (somewhere at the level of parsing a query maybe?).
Therefore, these `pgbuf_fix_old_and_check_repl_desync` will be called on both PTS and ATS - for whatever purpose - the forking between ATS and PTS is done inside the function. I am not able to answer myself why they are not called on the PS as well.
As for the vacuum part, I do not have an answer.

Reply:
After checking code, the WRITE latch in btree_find_oid_and_its_page  is correct.
As a general rule, the presence of BTREE_OP_PURPOSE  arguments suggests that function is called in a context where at least one page needs to be WRITE latched.
This PR is not concerned with refactoring/optimization of page latching, so it is enough.